### PR TITLE
Add CIS6 standard

### PR DESCRIPTION
**Pull request by DOBEN**
## Purpose

addresses https://github.com/Concordium/concordium-dapp-examples/issues/82

## Changes

- Add CIS6 standard.
- Fix sphinx docker container version to fix pipeline.

### CODE REVIEW

File: source/CIS/cis-6.rst [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+
+Abstract
+========
+
+A standard interface that defines the logged events for tracking items in a smart contract.
+The interface defines two events:
+
+- *ItemCreatedEvent* loggs the ``item_id``, ``metadata_url``, and the ``initial_status``;
+- *ItemStatusChangedEvent* loggs the ``item_id``, ``new_status``, and the ``additional_data``;
```
Comment (Bargsteen) [inline source/CIS/cis-6.rst:30]:
```suggestion
- *ItemCreatedEvent* logs the ``item_id``, ``metadata_url``, and the ``initial_status``;
- *ItemStatusChangedEvent* logs the ``item_id``, ``new_status``, and the ``additional_data``;
```

File: source/CIS/cis-6.rst [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+.. _CIS-6-ItemId:
+
+``ItemId``
+^^^^^^^^^^
+
+Each item MUST have a unique ``ItemId``.
+An ``ItemId`` is a variable-length ASCII string up to 255 characters.
+
+The identifier is serialized as: first 1 byte for the length followed this many bytes for the ASCII encoding of the identifier::
```
Comment (Bargsteen) [inline source/CIS/cis-6.rst:49]:
```suggestion
The identifier is serialized as: 1 byte for the length (``n``) followed by this many bytes for the ASCII encoding of the identifier::
```

File: source/CIS/cis-6.rst [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+                     | (1: Byte) (hash: Byte³²)
+
+  MetadataUrl ::= (n: Byte²) (url: Byteⁿ) (checksum: MetadataChecksum)
+
+.. _CIS-6-Status:
+
+``Status``
+^^^^^^^^^^
+
+The status defined by this specification are serialized using one byte to discriminate the different statuses that an item can have.
```
Comment (Bargsteen) [inline source/CIS/cis-6.rst:74]:
```suggestion
The status defined by this specification is serialized using one byte to discriminate the different statuses that an item can have.
```

File: source/CIS/cis-6.rst [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+It is serialized as: a byte of value (``n``)::
+
+  Status ::= (n: Byte)
+
+.. _CIS-6-AdditionalData:
+
+``AdditionalData``
+^^^^^^^^^^^^^^^^^^
+
+Additional bytes to include in the event ``ItemStatusChangedEvent``, which can be used to add additional usecase-specific data (e.g. temperature, longitude, latitude, ...) to the event.
```
Comment (Bargsteen) [inline source/CIS/cis-6.rst:86]:
```suggestion
Additional bytes to include in the event ``ItemStatusChangedEvent``, which can be used to add additional use-case-specific data (e.g. temperature, longitude, latitude, ...) to the event.
```

File: source/CIS/cis-6.rst [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+A custom event SHOULD NOT have a first byte colliding with any of the events defined by this specification.
+
+.. _CIS-6-events-ItemCreatedEvent:
+
+``ItemCreatedEvent``
+^^^^^^^^^^^^^^^^^^^^
+
+A ``ItemCreatedEvent`` event MUST be logged when a new item is created in the smart contract.
+
+The ``ItemCreatedEvent`` event is serialized as: first a byte with the value of 0, followed by the :ref:`CIS-6-ItemId` (``id``), the :ref:`CIS-6-MetadataUrl` (``metadata``), and then the :ref:`CIS-6-Status` (``initial_status``)::
```
Comment (Bargsteen) [inline source/CIS/cis-6.rst:105]: We are using `255 - n` for the other event ids (CIS-2, CIS-4, etc.). Why are we not doing it here?

**Review by Bargsteen — COMMENTED** [review]
Looks good! I just have a few questions ;)

File: source/CIS/cis-6.rst (hunk 5 above)
Comment (DOBEN) [reply source/CIS/cis-6.rst:105]: Makes sense:

Then CIS-5 should be updated to use the NonceEvent tag 250 and 243, 242, 241, 240, 239, 238 for the remaining 6 events.
And this CIS-6 standard will use the tags 237 and 236.

File: source/CIS/cis-6.rst [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+A URL and optional checksum for metadata stored outside of this contract.
+
+It is serialized as: 2 bytes for the length (``n``) of the metadata URL in little-endian and then this many bytes for the URL to the metadata (``url``) followed by an optional checksum.
+The checksum is serialized by 1 byte to indicate whether a hash of the metadata is included.
+If its value is 0, then there is no hash; if the value is 1, then 32 bytes for a SHA256 hash (``hash``) follows::
+
+  MetadataChecksum ::= (0: Byte)
+                     | (1: Byte) (hash: Byte³²)
+
+  MetadataUrl ::= (n: Byte²) (url: Byteⁿ) (checksum: MetadataChecksum)
```
Comment (limemloh) [inline source/CIS/cis-6.rst:67]: This standard should probably include some format information of the metadata stored at the url as well

File: source/CIS/cis-6.rst [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+
+``Status``
+^^^^^^^^^^
+
+The status defined by this specification is serialized using one byte to discriminate the different statuses that an item can have.
+The smart contract can have up to 255 different statuses defined for an item.
+
+It is serialized as: a byte of value (``n``)::
+
+  Status ::= (n: Byte)
```
Comment (limemloh) [inline source/CIS/cis-6.rst:79]: Should the metadata to include some name and description for each possible value of the status byte?

File: source/CIS/cis-6.rst [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+
+``ItemId``
+^^^^^^^^^^
+
+Each item MUST have a unique ``ItemId``.
+An ``ItemId`` is a variable-length ASCII string up to 255 characters.
+
+The identifier is serialized as: 1 byte for the length (``n``) followed by this many bytes for the ASCII encoding of the identifier::
+
+  StandardIdentifier ::= (n: Byte) (id: Byteⁿ)
```
Comment (limemloh) [inline source/CIS/cis-6.rst:51]:
```suggestion
  ItemId ::= (n: Byte) (id: Byteⁿ)
```

File: source/CIS/cis-6.rst (hunk 6 above)
Comment (DOBEN) [reply source/CIS/cis-6.rst:67]: The current scope of the standard was to define the events so we can have an `explorerTrackAndTrace` page in the block explorer. The metadata file is use case specific and could include different information. So far there hasn't been a strong requirement given to us to standardize the metadata file.

File: source/CIS/cis-6.rst (hunk 7 above)
Comment (DOBEN) [reply source/CIS/cis-6.rst:79]: The schema includes the human-readable representation (e.g. `InTransit`) of the meaning of the `Status` byte.

```
{
  "ItemStatusChanged": [
    {
      "additional_data": {
        "bytes": []
      },
      "item_id": "1600000000000000",
      "new_status": {
        "InTransit": []
      }
    }
  ]
}
```

**Review by limemloh — APPROVED** [review]

**Pull request merged**